### ISSUE TITLE
Suggested Changes to Edit Taxon Name and Related Data Page

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#3265]
From https://github.com/ArctosDB/arctos/issues/2695#issuecomment-676623830

See https://docs.google.com/document/d/1b44lTvw07ETmXjIj6CyJkREq1jnJxLjZtJ5EB8jrXbM/edit

**Comment by dustymc** [comment]
Everything MUST be data-driven. If I can add a relationship to the code table and your proposed UI accommodates it, then I have no objections. If I can't, then it's not acceptable.

The possibility that I'll add 200 relationships should be considered as well - whatever happens in the UI needs to deal with the scope of the data changing within realistic limits.

**Comment by Jegelewicz** [comment]
Let me in on where I am creating problems please!

I am testing as I go to make sure stuff "works", but it's difficult because this flips between test and production depending upon what is happening...

**Comment by Jegelewicz** [comment]
I do see this - which I don't think I changed? Something doesn't seem right here though:

<td>
  <select name="taxon_relationship" size="1" class="reqdClr">
	<cfloop query="ctRelation">
		<option <cfif ctRelation.taxon_relationship is relations.taxon_relationship>
		selected="selected" </cfif>value="#ctRelation.taxon_relationship#">#ctRelation.taxon_relationship#
		</option>
	</cfloop>
  </select>
</td>

**Comment by dustymc** [comment]
I was looking at your relationships mockup - not sure what you're planning, but that table could be built static. If that's not what you had in mind, ignore.

>flips between test and production 

We could do something radical in https://github.com/ArctosDB/arctos/issues/2769 - just decide we're going to release every tuesday night or something - but I think that would be a difficult/impossible model to live under.

https://github.com/ArctosDB/internal/issues/64 has the potential to fix this - just grab a copy of the DB and UI and install it on your laptop, merge when you've done what you're doing - which opens up huge security concerns even if it doesn't have the tech probs I suspect it does.

The /fix/ folder is the most realistic thing I've found for this, even if it's far from perfect.

**Comment by dustymc** [comment]
try three tics

```
this should be wysiwyg
```

or let me know file and line number

**Comment by Jegelewicz** [comment]
```
<td>
							<select name="taxon_relationship" size="1" class="reqdClr">
								<cfloop query="ctRelation">
									<option <cfif ctRelation.taxon_relationship is relations.taxon_relationship>
										selected="selected" </cfif>value="#ctRelation.taxon_relationship#">#ctRelation.taxon_relationship#
									</option>
								</cfloop>
							</select>
						</td>
```

**Comment by Jegelewicz** [comment]
It's that inside if statement that seems odd and turns red in Brackets.

**Comment by Jegelewicz** [comment]
> I was looking at your relationships mockup - not sure what you're planning, but that table could be built static. If that's not what you had in mind, ignore.

I tried to just change your list relationship display into a table to make things visually consistent. I think I succeeded in getting the loop to run so that you can have as many relationships as you like....but only testing will tell!

**Comment by dustymc** [comment]
That seems right to me....

**Comment by Jegelewicz** [comment]
> We could do something radical in #2769 - just decide we're going to release every tuesday night or something - but I think that would be a difficult/impossible model to live under.

Agree. I just want to make sure that some crazy thing I am working with doesn't end up in production before it is fully baked. I can use my sandbox in /fix, but that doesn't allow testing for functionality (at least it didn't for the loan search stuff). I hate to do a bunch of UI stuff only to find that I have broken something and I'm not sure what did it. I'm just learning about what happens with a # in a link!

I figured if I had something listed as "Active Development" then you would know to warn me before you pull anything to production? Also, I should probably save all the code to someplace before I start editing so that I can easily just get the original back. (Or you could if I happen to be unavailable).

**Comment by Jegelewicz** [comment]
> That seems right to me....

Then I will just keep on keeping on unless I notice some kind of problem.

**Comment by dustymc** [comment]
>doesn't allow testing for functionality

Yea best case it'll require replacing all ``/thisTool.cfm`` with a local link and then reversing that, and taxonomy is weirder because it uses "nice" URLs. Ain't ideal....

>something listed as "Active Development" then you would know to warn me

I just (occasionally) close those after I push to production.... If you want to suggest something better I'll try to follow it - and if we do get something worked out it should go to https://handbook.arctosdb.org/how_to/developer-guide.html

>save all the code 

You can dig it out of the last branch, but yea if I'm doing something too crazy I often end up with a bunch of 'version' text files or the whole previous 'version' in a comment or something just to make things easy.

**Comment by Jegelewicz** [comment]
> the whole previous 'version' in a comment or something

Yeah - that's what I should do.

> I just (occasionally) close those after I push to production.... If you want to suggest something better I'll try to follow it - and if we do get something worked out it should go to https://handbook.arctosdb.org/how_to/developer-guide.html

If I use the method of commenting out the old while I work, then I can always comment out my work and replace with the old if I am not actively working or if you are going to pull to production. All I would need is a little notice when you are going to do that so that I can make sure I'm not right in the middle of something.

**Comment by Jegelewicz** [comment]
@dustymc here is one I am struggling with. Currently we just have this text for deleting a pub from a taxon name:

```
<td>
        <a href="editTaxonomy.cfm?action=removePub&taxonomy_publication_id=#taxonomy_publication_id#&taxon_name_id=#taxon_name_id#">Remove</a>
</td>
```

Can we make that look like a button? My attempt:

```
<td>
       <input type="button" value="Remove" class="delBtn" onclick="editTaxonomy.cfm? action=removePub&taxonomy_publication_id=#taxonomy_publication_id#&taxon_name_id=#taxon_name_id#">
</td>
```
doesn't work.

**Comment by dustymc** [comment]
There are (at least) three possibilities.

1. Link - it's sorta weird because it DOES stuff, not GOES PLACES here.
2. Form - expected behavior here, but it's more code and previous to HTML5 there were nesting issues
3. Something hacky, like making a link look like a button or a button act like a link. 

I'm not suggesting we need to sort that out now, but the deeper we dig this hole..... Would be a useful section of https://handbook.arctosdb.org/how_to/developer-guide.html, so at least I wouldn't have any good excuses for doing it "wrong."

(2) is almost certainly the 'correct' approach.

(3) in the form of a new "lookslikeabutton" css class or something isn't any more wrong than it is now, and might look better.

(3) in the form of a button that doesn't act like a button could be accomplished with window.location, which ALMOST works like a GET anyway. 

tl;dr: onClick is a JS event, it'll only execute JS functions

```
<input type="button" value="Remove" class="delBtn" onclick="window.location='editTaxonomy.cfm?action=removePub&taxonomy_publication_id=#taxonomy_publication_id#&taxon_name_id=#taxon_name_id#';">
```

**Comment by dustymc** [comment]
@Jegelewicz I'm 99% sure I'm going to have a release ready tonight, if you've got anything you need to hide under the rug first.

FWIW I very much dislike having to click around on the thing I already clicked to in order to do anything. The panes are cool, but they should pop open until my screen is full, or remember the last state I put them in, or start open and let me close them (and remember that), or SOMETHING that doesn't mean a minimum of 5 extra clicks to just see what's already there. I'm not much of a user so if that's just me it's fine, but I think the approach should be discussed before you go too much further, and that page is a nice demo on which to base that discussion.

**Comment by Jegelewicz** [comment]
> I'm 99% sure I'm going to have a release ready tonight, if you've got anything you need to hide under the rug first.

I missed that! Let's see who notices....

> The panes are cool, but they should pop open until my screen is full, or remember the last state I put them in, or start open and let me close them (and remember that), or SOMETHING that doesn't mean a minimum of 5 extra clicks to just see what's already there. I'm not much of a user so if that's just me it's fine, but I think the approach should be discussed before you go too much further, and that page is a nice demo on which to base that discussion.

@dustymc we discussed this in the taxonomy committee meeting yesterday. These would be the most likely users of this form page and they seemed to like the set-up that eliminates a bunch of scrolling. We did talk about the order of the things and I will probably change that. One thing I need to figure out is how to have the page remember what was open after certain things are "done" like adding or removing a pub, because no one will want to have to re-open that pane every time they save something. Since I'm learning as I go - that might take me a day or even two!

I figured since this isn't the most heavily used edit page it is a good place to test some of this. I really did base my changes upon things people have said about UI.

 - I have to scroll way too much
 - I just want to see what I am doing on one page
 - We need consistency in the presentation of buttons and links
 - I want things clean and simple as possible (HAHAHAHAHA!!! but I also WANT IT ALL)

**Comment by Jegelewicz** [comment]
Comments on updated page from taxonomy committee:

 - Change "Edit Name" to "Edit Name and Name Type" and move this down since it isn't often used.
 - buttons to "Do" things and [ links ] to "Go Places"
 - return to open pane after doing anything

**Comment by dustymc** [comment]
Let's take this to https://github.com/ArctosDB/arctos/issues/3310
